orx/grantlee/0.2/OrxJs.qs

```diff
@@ -0,0 +1,272 @@
+var buffer = '';
+
+function append() {
+    for (var i = 0; i < arguments.length; i++) {
+        if (arguments[i] !== undefined) {
+            buffer += arguments[i] + '\n';
+        }
+    }
+    if (arguments.length == 0) {
+        buffer += '\n';
+    }
+}
+
+function friendlyName(name) {
+    name = name.replace(/[^a-z0-9]/gi, ' ');        // Keep alpha-numeric
+    name = name[0].toUpperCase() + name.slice(1); // Proper case
+    return name;
+}
+
+function getId(name) {
+    // 'big enemy/hard-boss/boss_01' -> ['big', 'enemy', 'hard', 'boss', 'boss', '01']
+    var parts = name.split(/\/|_|-| /g);
+    
+    for (var i = 0; i < parts.length; i++) {
+        parts[i] = friendlyName(parts[i]);
+    }
+    
+    // Optimize any sequences of the same identifier.
+    // E.g. ['Boss', 'Boss', 'Boss'] -> 'Boss'
+    var clean = parts[0];
```

Comment:
By doing this simplification, you might end up with collisions. Based on your example, consider:
`big enemy/hard-boss/boss_01`
`big enemy/hard/boss_01`

Reply:
Nice catch :) I will try to improve the algorithm. Though, when reducing the names there will always be edge cases. I will add a setting to toggle the generation of "friendly names".